python/az/aro/azext_aro/custom.py

```diff
@@ -209,3 +286,42 @@ def generate_random_id():
                  ''.join(random.choice('abcdefghijklmnopqrstuvwxyz1234567890')
                          for _ in range(7)))
     return random_id
+
+
+def get_route_tables_from_subnets(cli_ctx, subnets):
+    network_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK)
+
+    route_tables = set()
+    for sn in subnets:
+        sid = parse_resource_id(sn)
+
+        subnet = network_client.subnets.get(resource_group_name=sid['resource_group'],
+                                            virtual_network_name=sid['name'],
+                                            subnet_name=sid['resource_name'])
+
+        if subnet.route_table is not None:
+            route_tables.add(subnet.route_table.id)
+
+    return route_tables
+
+
+def get_cluster_network_resources(cli_ctx, oc):
+    master_subnet = oc.master_profile.subnet_id
+    worker_subnets = {w.subnet_id for w in oc.worker_profiles}
+
+    master_parts = parse_resource_id(master_subnet)
+    vnet = resource_id(
+        subscription=master_parts['subscription'],
+        resource_group=master_parts['resource_group'],
+        namespace='Microsoft.Network',
+        type='virtualNetworks',
+        name=master_parts['name'],
+    )
+
+    route_tables = get_route_tables_from_subnets(cli_ctx, worker_subnets | {master_subnet})
+
+    resources = set()
+    resources.add(vnet)
+    resources.update(route_tables)
+
+    return resources
```

Comment:
this code is duplicated in aro_create - I wouldn't mind seeing a separate function (DRY)

Reply:
bump :-)